infra/bridge/src/controllers/promotions.js

```diff
@@ -68,57 +74,28 @@ const persistEvent = async ({
       .digest('hex')
   }
 
-  const txn = await db.sequelize.transaction()
-
   try {
+    const profileData = getUserProfileFromEvent({ event, socialNetwork, type })
     await GrowthEvent.insert(
       logger,
-      1,
+      1, // insert a single entry
       identity,
       PromotionEventToGrowthEvent[socialNetwork][type],
-      contentHash,
-      event,
+      contentHash, // set customId to the content hash.
+      profileData, // store the profile data that contains the user's social stats in the GrowthEvent.data column.
```

Comment:
Can we store both `event` and `profileData` here? Or persist the `event` somewhere else?

We do get stuffs like tweet/status id from the `event` that can be useful during verification before payout (We don't have search through the user's timeline again, we could just fetch that specific tweet and see it is not a 404). Also, it will be useful in debugging if anything goes wrong

Reply:
Sounds good. It is duplicate data but could be useful for debugging so let's store it for now and we can always stop storing it in future if it takes too much space.